azure-functions/acbs-function/acbs-amend-facility/index.js

```diff
@@ -0,0 +1,157 @@
+      const hasDeal = amendment.deal && amendment.deal.dealSnapshot;
+
+      // Payload verification
+      if (hasFacilityId && hasAmendment && hasFacility && hasDeal) {
+        const { facility, deal, facilityId } = amendment;
+
+        // 1. DAF : activity-get-facility-master: Retrieve ACBS `Facility Master Record` with eTag
+        const { acbsFacility: fmr, etag } = yield context.df.callActivityWithRetry('activity-get-facility-master', retryOptions, { facilityId });
+
+        if (!!fmr && !!etag && !!facility.facilitySnapshot) {
+          let facilityMasterRecordAmendments;
+          let facilityLoanRecordAmendments;
+          const amendments = {
+            amendment,
+          };
+          const { facilitySnapshot } = facility;
+
+          /**
+           * *************************** AMENDMENT MAPPING ***************************
+           */
+
+          // 2.1. Facility Master Record (FMR)
+          const fmrMapped = mappings.facility.facilityMasterAmend(fmr, amendments, deal);
+
+          // 2.2. FMR update
+          // 2.2.1 - UKEF Exposure
+          if (amendment.amount) {
+            const result = yield context.df.callActivityWithRetry('activity-update-facility-master', retryOptions, {
+              facilityId,
+              acbsFacilityMasterInput: fmrMapped,
+              updateType: 'amendAmount',
+              etag,
+            });
+
+            facilityMasterRecordAmendments = {
+              amount: {
+                ...result,
+              },
+            };
+          }
+
+          // 2.2.2. DAF : activity-get-facility-master: Retrieve ACBS `Facility Master Record` with new eTag
+          const updatedFmr = yield context.df.callActivityWithRetry('activity-get-facility-master', retryOptions, { facilityId });
+
+          // 2.2.3 - Cover end date
+          if (amendment.coverEndDate && updatedFmr.etag) {
+            const result = yield context.df.callActivityWithRetry('activity-update-facility-master', retryOptions, {
+              facilityId,
+              acbsFacilityMasterInput: fmrMapped,
+              updateType: 'amendExpiryDate',
+              etag: updatedFmr.etag,
+            });
+
+            facilityMasterRecordAmendments = {
+              ...facilityMasterRecordAmendments,
+              coverEndDate: {
+                ...result,
+              },
+            };
+          }
+
+          // 2.3. Facility Loan Record (FLR)
+          const flrMApped = mappings.facility.facilityLoanAmend(amendments, facility, fmr);
+
+          // 2.3 FLR update
+          // 2.3.1 Extract loan id for facility id
+          const loanId = yield context.df.callActivityWithRetry('activity-get-loan-id', retryOptions, {
+            facilityId,
+          });
+
+          if (loanId) {
+            // 2.3.2 - UKEF Exposure
+            // Loan amount - `Bond` facility types only
+            if (amendment.amount && facilitySnapshot.type === CONSTANTS.FACILITY.FACILITY_TYPE.BOND) {
```

Comment:
is it possible to deconstruct { FACILITY } = require('___/constants); or even {FACILITY_TYPE} = require('___/constants/facility);

Reply:
Updated.